osquery/tables/system/windows/tpm_info.cpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/core/tables.h>
+#include <osquery/core/windows/wmi.h>
+#include <osquery/logger/logger.h>
+
+namespace osquery {
+namespace tables {
+
+QueryData genTpmInfo(QueryContext& context) {
+  QueryData resultsdata;
+  std::string wmiclass{"SELECT * FROM Win32_Tpm"};
+
+  BSTR wminamespace = ::SysAllocString(L"root\\cimv2\\Security\\MicrosoftTpm");

```

Comment:
ca/should the namespace string be a constant?

Reply:
BSTRs are a special system type, unfortunately although they're typedef'd to `wchar_t *`. They should always be allocated with `SysAllocString` and freed with `SysFreeString`. BSTRs use a different memory allocator, even (https://docs.microsoft.com/en-us/previous-versions/windows/desktop/automat/bstr).